bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/NetUtil.java

```diff
@@ -41,28 +49,52 @@
  * @author Markus Rathgeb - Initial contribution and API
  * @author Mark Herwege - Added methods to find broadcast address(es)
  * @author Stefan Triller - Converted to OSGi service with primary ipv4 conf
+ * @author Gary Tse - Network address change listener
  */
 @Component(configurationPid = "org.eclipse.smarthome.network", property = { "service.pid=org.eclipse.smarthome.network",
         "service.config.description.uri=system:network", "service.config.label=Network Settings",
-        "service.config.category=system" })
+        "service.config.category=system", "service.config.network.interface_poll.frequency_seconds=120" })
 @NonNullByDefault
 public class NetUtil implements NetworkAddressService {
 
     private static final String PRIMARY_ADDRESS = "primaryAddress";
     private static final String BROADCAST_ADDRESS = "broadcastAddress";
+    private static final String INTERFACE_POLL_FREQUENCY_SECONDS_KEY = "service.config.network.interface_poll.frequency_seconds";
+    private static final String THREAD_POOL_NAME = "common";
```

Comment:
If we introduce something like a "common" thread pool, we should NOT declare it within NetUtil, but probably rather in ThreadPoolManager. If you move it there, it might be worthwhile to add some JavaDoc to the constant. (@maggu2810, would love your input on whether you think a "common" thread pool is at all desired - I think it can make sense for "small" pieces of logic that want to schedule something and where it does not make sense to setup any dedicated pool or permanent thread).